collect_app/src/main/java/org/odk/collect/android/preferences/ServerPreferencesFragment.java

```diff
@@ -99,10 +94,46 @@ public static ServerPreferencesFragment newInstance(boolean adminMode) {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         addPreferencesFromResource(R.xml.server_preferences);
-
         initProtocolPrefs();
     }
 
+    @Override
+    public void onDisplayPreferenceDialog(Preference preference) {
+        if (preference instanceof CustomEditTextPreference) {
+            switch (preference.getKey()) {
+                case "server_url":
+                    dialogFragment = ServerUrlEditTextPreferenceDialog.newInstance(preference.getKey());
```

Comment:
Not sure why you have to create your own dialogs for these preferences (and not just use `EditTextPreference`)? Have you looked through the docs at https://developer.android.com/guide/topics/ui/settings/ to see if there are other ways of implementing the features there is code for in the dialogs?

Reply:
`EditTextPreference.getEditText()` method was removed in `androidx.preference` as discussed [here](https://stackoverflow.com/questions/32477726/android-no-getedittext-method-in-edittextpreference-with-preference-support-li). That's why I used `CustomEditTextPreference`. I do see that now that there's an easy way of doing so. I will review my code once again and then will ask for re-review.

Reply:
Sorry for the redundant code. I removed the `CustomEditTextPreference` class. I followed [this article](https://medium.com/@JakobUlbrich/building-a-settings-screen-for-android-part-3-ae9793fd31ec) for creating Reference layer dialog and Reset dialog.